department-of-veterans-affairs/vets-website
# [MHV-52990] Bug: Start a new message, radio buttons

## Summary

- The category selection within the Start a new message have different name attributes per radio button.
- This update corrects the category radio buttons' name so that all options have the same name.
- Screen-readers should be able to navigate to each radio button and announce the group association "1 of 6", making it clear to users that they need to choose one option from the presented set.

## Related issue(s)

### [MHV-52990](https://jira.devops.va.gov/browse/MHV-52990) Bug: Start a new message, radio buttons


> ### Description:
> The category selection within the Start a new message have different name attributes per radio button.

> The name attribute is crucial for grouping radio buttons together, especially for form submissions and accessibility. When radio buttons share the same name attribute, they are considered part of the same group. This behavior ensures that a person's selection is clear and that the choices are mutually exclusive within the specified group.

> For people who rely on screen readers or other assistive technologies, having a consistent and meaningful grouping is essential for understanding the structure and purpose of the form. If radio buttons with different names are used for options that should be mutually exclusive, screen readers may not interpret the relationship correctly, leading to confusion for users with visual impairments.
When radio buttons are correctly grouped using the name attribute, screen readers announce the group association "1 of 6", making it clear to people that they need to choose one option from the presented set. This aids in providing a more accessible and understandable user experience for individuals with disabilities. Therefore, using the same name attribute for radio buttons that represent mutually exclusive choices is not just a convention for developers but is also an accessibility best practice.

> ### *Acceptance Criteria: *
> Have the same Name attributes match such as name=”category”
Screen readers read this as a group “1 of 6”

> ### Steps to Reproduce:
> Log into Secure messages
Navigate to Inbox
Click “Start a new message”
Click “Continue to start message”
Start screen reader
Navigate to Category field
Navigate through the radio buttons

> ### Expected Behavior:
> Expected the Screen reader to read these as radio button 1 of 6 (or however many options there are) when navigating to the first radio button. Then when navigating to the second radio button, expected to hear 2 of 6, and so on.

> Accessibility Specialist: @BobbyBaileyRB 
>

## Testing done

- _Unit test added to capture the name of categories radio buttons_
- _Manual testing with a screen-reader_
- _Used Axe Dev Tool to run checks_
- _Screenreader should announce radio button option as part of a set, i.e. "1 of 6", "2 of 6" , etc._


## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*Va.gov - Secure Messaging*

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
